### PR TITLE
change authentication logic to be handled by back-end

### PR DESCRIPTION
**Pull request by nazaninreihani**

### CODE REVIEW

File: www/js/share/components/check-user-status/check-user-status.controller.js [hunk 1]
```diff
@@ -257,6 +247,8 @@
         $scope.$on("get_account_status", (e, get_account_status) => {
             if (get_account_status.hasOwnProperty("status")) {
                 vm.status = get_account_status.status;
+                vm.promptClientToAuthenticate = get_account_status.prompt_client_to_authenticate;
```
Comment (morteza-binary) [inline www/js/share/components/check-user-status/check-user-status.controller.js]: @nazaninreihani  It does not need to add the value to vm object if you don't need it more.

File: www/js/share/components/check-user-status/check-user-status.controller.js [hunk 2]
```diff
@@ -179,19 +178,10 @@
...
-        vm.authenticateStatus = function(status) {
-            vm.authenticated = status.indexOf("authenticated") > -1;
-            if (
-                !vm.authenticated &&
-                (vm.isFinancial ||
-                    (vm.isCR && (vm.balance >= 200 || vm.mt5LoginList.length > 0)) ||
-                    vm.isMLT ||
-                    vm.isMX)
-            ) {
-                if (!appStateService.hasAuthenticateMessage) {
```
Comment (morteza-binary) [inline www/js/share/components/check-user-status/check-user-status.controller.js]: @nazaninreihani 
Do you handle that this part of code is execute just one time for each user authentication process?

As you know if our app try to keep itself connect, so it is possible to have multiple authentication process for an account during the runtime and in this way you push same messages in your messages list.

I think you should keep this condition in your new code, unless you handle it some where else.

Comment (nazaninreihani) [inline www/js/share/components/check-user-status/check-user-status.controller.js]: I forgot to put this condition in new changes :-D thanks

Comment (morteza-binary) [inline www/js/share/components/check-user-status/check-user-status.controller.js]: @nazaninreihani  NP

**Pull request merged**